docker-compose.yml

```diff
@@ -1,26 +1,29 @@
-version: "2"
+version: "2.3"
+
+x-env-mapping: &env
+  environment:
+    - CELERY_BROKER_URL=amqp://olympia:olympia@rabbitmq/olympia
+    - CELERY_RESULT_BACKEND=redis://redis:6379/1
+    - DATABASE_URL=mysql://root:@mysqld/olympia
+    - ELASTICSEARCH_LOCATION=elasticsearch:9200
+    - MEMCACHE_LOCATION=memcached:11211
+    - MYSQL_DATABASE=olympia
+    - MYSQL_ROOT_PASSWORD=docker
+    - OLYMPIA_SITE_URL=http://olympia.dev
+    - PYTHONDONTWRITEBYTECODE=1
+    - PYTHONUNBUFFERED=1
+    - RECURSION_LIMIT=10000
+    - REDIS_LOCATION=redis://redis:6379/0?socket_timeout=0.5
+    - TERM=xterm-256color
```

Comment:
This might need a short announcement that we now require at least docker-compose version 1.7 and related docker version. I hope this doesn't bring too many problems to contributors but we'll see.

Reply:
What problems do you think could arise?

Reply:
Well, I'm not sure what most distributions offer as the latest versions via their package managers so being almost-bleeding-edge usually isn't easy. But I think we should be safe. Just have to keep that in mind.